sklearn/cluster/_kmeans.py

```diff
@@ -1782,27 +1824,35 @@ def partial_fit(self, X, y=None, sample_weight=None):
         -------
         self
         """
+        is_first_call_to_partial_fit = not hasattr(self, 'cluster_centers_')
 
-        X = check_array(X, accept_sparse="csr", order="C",
-                        dtype=[np.float64, np.float32])
-        n_samples, n_features = X.shape
-        if hasattr(self.init, '__array__'):
-            self.init = np.ascontiguousarray(self.init, dtype=X.dtype)
+        X = self._validate_data(X, accept_sparse='csr',
+                                dtype=[np.float64, np.float32],
+                                order='C', accept_large_sparse=False,
+                                reset=is_first_call_to_partial_fit)
 
-        if n_samples == 0:
+        if X.shape[0] == 0:
             return self
 
+        self.random_state_ = getattr(self, "random_state_",
+                                     check_random_state(self.random_state))
         sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype)
 
         x_squared_norms = row_norms(X, squared=True)
-        self.random_state_ = getattr(self, "random_state_",
-                                     check_random_state(self.random_state))
-        if (not hasattr(self, 'counts_')
-                or not hasattr(self, 'cluster_centers_')):
-            # this is the first call partial_fit on this object:
+
+        if is_first_call_to_partial_fit:
+            # this is the first call to partial_fit on this object
+            self._check_params(X)
+
+            # Validate init array
+            init = self.init
+            if hasattr(init, '__array__'):
+                init = check_array(init, dtype=X.dtype, copy=True, order='C')
```

Comment:
codecov reports that this is not covered

Reply:
I added a test